File: e270_transpose_text/php/Transpose.php
<?php

$fileName = $argv[1];

$content = file_get_contents($fileName);

$lines = explode("\n", $content);

// determine maxLength
$maxLength = 0;
foreach ($lines as $line) {
    if (strlen($line) > $maxLength) {
        $maxLength = strlen($line);
    }
}

// Transpose
$result = [];
for ($i = 0; $i < $maxLength; $i++) {
    $tranposed = "";
    foreach ($lines as $line) {
        $tranposed .= ($i < strlen($line)) ? $line[$i] : ' ';
    }
    $result[] = $tranposed;
}

echo implode("\n", $result);

File: e270_transpose_text/java/Transpose.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Transpose
{
    public static void main (String[] args) throws java.lang.Exception
    {
        ArrayList<String> lines = new ArrayList<>();
        BufferedReader br = new BufferedReader(new FileReader(new File(args[0])));
        String in;
        int maxLength = 0;

        // Read the file into lines and determine the maxLength
        while ((in = br.readLine()) != null) {
            maxLength = (in.length() > maxLength) ? in.length() : maxLength;
            if (in.isEmpty()) {
                lines.add(" ");
            } else {
                lines.add(in);
            }
        }

        // Transpose
        ArrayList<String> result = new ArrayList<>();
        for (int i = 0; i < maxLength; i++) {
            String transposed = "";
            for (String line : lines) {
                transposed += (i < line.length() ? line.charAt(i) : ' ');
            }
            result.add(transposed.replaceAll("\\s+$", ""));
        }

        // Print Result
        for (String line: result) {
            System.out.println(line);
        }
    }
}